lib/io/can.cpp

```diff
@@ -1,8 +1,73 @@
 #include "can.hpp"
 
+#include <stdio.h>
+#include <unistd.h>
+
+#include <linux/can.h>
+#include <net/if.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+
 namespace hyped::io {
-Can::Can()
+Can::Can(hyped::core::ILogger &logger) : logger_(logger)
```

Comment:
I think you should extract the I/O out of the constructor into a function, e.g. `Can::initialise`.

The reason is the following: I/O may fail and you want to indicate that to your callee. No, logging doesn't count! Thus have a function that returns an enum value `kFailure` or `kSuccess`. In fact, this type has become so common that we might want to move it into core @kshxtij . But then again, they don't need to be interchangable so maybe not?

Reply:
Yeah i was thinking of creating a core::types::WriteResult enum with kFailure and kSucess while doing #19 anyways

Reply:
Done in #19, will retroactively fix everything else once that's merged